xbmc/Application.cpp

```diff
@@ -1123,8 +1124,22 @@ bool CApplication::Initialize()
   g_curlInterface.Unload();
 
   // initialize (and update as needed) our databases
-  CSplash::GetInstance().Show(g_localizeStrings.Get(24150));
-  CDatabaseManager::GetInstance().Initialize();
+  std::atomic<bool> notFinished = true;
+  CJobManager::GetInstance().Submit([&notFinished]() {
+    CDatabaseManager::GetInstance().Initialize();
+    notFinished = false;
+  });
+  std::string localizedStr = g_localizeStrings.Get(24150);
+  while (notFinished)
+  {
+    for (int i = 1; i <= 3 && notFinished; ++i)
+    {
+      if (CDatabaseManager::GetInstance().m_bIsUpgrading)
+        CSplash::GetInstance().Show(std::string(i, ' ') + localizedStr + std::string(i, '.'));
+      if (notFinished)
```

Comment:
I added it because the work might end while drawing the message and might avoid to wait 1 sec in that case.. not sure if it's worth it